ddtrace/http/headers.py

```diff
@@ -0,0 +1,81 @@
+import re
+from ..utils.http import normalize_header_name
+
+
+REQUEST = 'request'
+RESPONSE = 'response'
+
+NORMALIZE_PATTERN = re.compile(r'([^a-z0-9])+')
+
+
+def store_request_headers(headers, span, integration_config):
+    """
+    Store request headers as a span's tags
+    :param headers: All the request's http headers, will be filtered through the whitelist
+    :type headers: dict
+    :param span: The Span instance where tags will be store
+    :type span: ddtrace.Span
+    :param integration_config: An integration specific config object.
+    :type integration_config: ddtrace.settings.IntegrationConfig
+    """
+    _store_headers(headers, span, integration_config, REQUEST)
+
+
+def store_response_headers(headers, span, integration_config):
+    """
+    Store response headers as a span's tags
+    :param headers: All the response's http headers, will be filtered through the whitelist
+    :type headers: dict
+    :param span: The Span instance where tags will be store
+    :type span: ddtrace.Span
+    :param integration_config: An integration specific config object.
+    :type integration_config: ddtrace.settings.IntegrationConfig
+    """
+    _store_headers(headers, span, integration_config, RESPONSE)
+
+
+def _store_headers(headers, span, integration_config, request_or_response):
+    """
+    :param headers: A dict of http headers to be stored in the span
+    :type headers: dict
+    :param span: The Span instance where tags will be store
+    :type span: ddtrace.span.Span
+    :param integration_config:
+    :param integration_config: An integration specific config object.
+    :type integration_config: ddtrace.settings.IntegrationConfig
+    """
+    if not isinstance(headers, dict):
```

Comment:
Curious thought. Some places, like WSGI use a list of tuples for headers.

`[('Content-Length', 11), ]`

I wonder if we will encounter that here at all?

Reply:
For now I adapted the values in the integration itself, but we can make this method more permissive. E.g. https://github.com/DataDog/dd-trace-py/pull/647/files#diff-b7b923cbf57be67da4667519da874510R44

Reply:
yeah, I think I am all for making this method more permissive, dict vs list of tuples vs iterable of tuples vs etc.

All should be easy to handle here, and saves us from having to always cast to `dict` when calling.